plugins/module_utils/dependency_checker.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) IBM Corporation 2025
+# Licensed under the Apache License, Version 2.0
+
+from __future__ import absolute_import, division, print_function
+import re
+import subprocess
+import sys
+import json
+
+__metaclass__ = type
+
+# ------------------------------
+# Compatibility Matrix
+# ------------------------------
+COMPATIBILITY_MATRIX = [
+    {"zoau_version": "1.3.5", "python_version": "3.12", "galaxy_core_version": "1.12.0", "zos_range": "2.5-3.1"},
+    {"zoau_version": "1.3.5.1", "python_version": "3.10", "galaxy_core_version": "1.12.0", "zos_range": "2.5-3.1"},
+    {"zoau_version": "1.3.6.0", "python_version": "3.10", "galaxy_core_version": "1.10.1", "zos_range": "2.5-3.1"},
+    {"zoau_version": "1.4.0", "python_version": "3.11", "galaxy_core_version": "1.11.0", "zos_range": "2.5-3.1"},
+    {"zoau_version": "1.4.0", "python_version": "3.11", "galaxy_core_version": "1.12.0", "zos_range": "2.5-3.1"},
+]
+
+# ------------------------------
+# Helpers
+# ------------------------------
+def run_command(module, cmd):
+    rc, out, err = module.run_command(cmd)
+    if rc != 0:
+        module.fail_json(msg=f"Command failed: {cmd}", stdout=out, stderr=err)
+    return out.strip()
+
+# ------------------------------
+# Version Fetchers
+# ------------------------------
+def get_zoau_version(module):
+    output = run_command(module, "zoaversion")
+    match = re.search(r'v(\d+\.\d+\.\d+(?:\.\d+)?)', output)
+    return match.group(1) if match else None
+
+def get_python_version_info():
+    return sys.version_info.major, sys.version_info.minor
+
+def get_python_version():
+    return f"{sys.version_info.major}.{sys.version_info.minor}.0"
+
+def get_zos_version(module):
+    output = run_command(module, "zinfo -t sys -j")
+    try:
+        data = json.loads(output)
+        sys_info = data.get("data", {}).get("sys_info", {})
+        version = sys_info.get("product_version")
+        release = sys_info.get("product_release")
+        if version and release:
+            return f"{int(version)}.{int(release)}"
+    except json.JSONDecodeError:
+        match_ver = re.search(r'"product_version":"(\d+)"', output)
+        match_rel = re.search(r'"product_release":"(\d+)"', output)
+        if match_ver and match_rel:
+            return f"{int(match_ver.group(1))}.{int(match_rel.group(1))}"
+    return None
+
+def get_galaxy_core_version():
```

Comment:
If we are in the managed node we cannot access the collection version using `ansible-galaxy` command, this is what we were talking about the other day, is it possible to get through an import by fetching the valye from `meta/ibm_zos_core`?